jhipster-framework/src/test/java/io/github/jhipster/web/util/PaginationUtilUnitTest.java

```diff
@@ -40,171 +41,142 @@
         assertNotNull(strHeaders);
-        assertTrue(strHeaders.size() == 1);
+        assertEquals(1, strHeaders.size());
         headerData = strHeaders.get(0);
-        assertTrue(headerData.split(",").length == 3);
-        expectedData = "</api/_search/example?page=6&size=50&query=Test1%2C+test2>; rel=\"prev\","
-            + "</api/_search/example?page=7&size=50&query=Test1%2C+test2>; rel=\"last\","
-            + "</api/_search/example?page=0&size=50&query=Test1%2C+test2>; rel=\"first\"";
+        assertEquals(3, headerData.split(",").length);
+        expectedData = "</api/_search/example?query=Test1%2C%20test2&page=6&size=50>; rel=\"prev\"," +
+            "</api/_search/example?query=Test1%2C%20test2&page=7&size=50>; rel=\"last\"," +
+            "</api/_search/example?query=Test1%2C%20test2&page=0&size=50>; rel=\"first\"";
         assertEquals(expectedData, headerData);
     }
 
     @Test
     public void greaterSemicolonTest() {
-        mockRequest.setParameter("query", "Test>;test");
-        String baseUrl = "/api/_search/example";
-        List<String> content = new ArrayList<>();
-        Page<String> page = new PageImpl<>(content);
-        String query = "Test>;test";
-        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(mockRequest, page);
+        uriBuilder.queryParam("query", "Test>;test");
+        Page<String> page = new PageImpl<>(new ArrayList<>());
+        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(uriBuilder, page);
         List<String> strHeaders = headers.get(HttpHeaders.LINK);
         assertNotNull(strHeaders);
-        assertTrue(strHeaders.size() == 1);
+        assertEquals(1, strHeaders.size());
         String headerData = strHeaders.get(0);
-        assertTrue(headerData.split(",").length == 2);
+        assertEquals(2, headerData.split(",").length);
         String[] linksData = headerData.split(",");
-        assertTrue(linksData.length == 2);
-        assertTrue(linksData[0].split(">;").length == 2);
-        assertTrue(linksData[1].split(">;").length == 2);
-        String expectedData = "</api/_search/example?page=0&size=0&query=Test%3E%3Btest>; rel=\"last\","
-            + "</api/_search/example?page=0&size=0&query=Test%3E%3Btest>; rel=\"first\"";
+        assertEquals(2, linksData.length);
+        assertEquals(2, linksData[0].split(">;").length);
+        assertEquals(2, linksData[1].split(">;").length);
+        String expectedData = "</api/_search/example?query=Test%3E%3Btest&page=0&size=0>; rel=\"last\"," +
+            "</api/_search/example?query=Test%3E%3Btest&page=0&size=0>; rel=\"first\"";
         assertEquals(expectedData, headerData);
         List<String> xTotalCountHeaders = headers.get("X-Total-Count");
-        assertTrue(xTotalCountHeaders.size() == 1);
-        assertTrue(Long.valueOf(xTotalCountHeaders.get(0)).equals(0L));
-    }
-
-    @Test
-    public void sortAndCriteriaTest() {
```

Comment:
I didn't see the added value of this test so I removed it.

Reply:
this is to test the bug we had previously, you were losing the sort and criteria when navigating to another page

Reply:
Yes but the fix uses a generic way to add the query params. So it's already tested in the previous tests